Sources/Sentry/include/SentryProfiler.h

```diff
@@ -48,35 +44,19 @@ SENTRY_EXTERN_C_END
 
 @interface SentryProfiler : NSObject
 
-/**
- * Start the profiler, if it isn't already running, for the span with the provided ID. If it's
- * already running, it will track the new span as well.
- */
-+ (void)startForSpanID:(SentrySpanId *)spanID hub:(SentryHub *)hub;
-
-/**
- * Report that a span ended to the profiler so it can update bookkeeping and if it was the last
- * concurrent span being profiled, stops the profiler.
- */
-+ (void)stopProfilingSpan:(id<SentrySpan>)span;
+/** Start the profiler, if it isn't already running. */
++ (void)startWithHub:(SentryHub *)hub;
```

Comment:
`m`: I would prefer not to pass in the hub, but only the things we need from the hub, which aren't that many.

Reply:
I agree it's not ideal, but we actually already do it here: https://github.com/getsentry/sentry-cocoa/blob/04beb80384b82c9d19b4b849d98fc820fcca83de/Sources/Sentry/include/SentryProfiler.h#L53. This just moved the argument to a different method. Note that we also pass a reference to the hub to SentryTracer: https://github.com/getsentry/sentry-cocoa/blob/04beb80384b82c9d19b4b849d98fc820fcca83de/Sources/Sentry/include/SentryTracer.h#L115-L116

I agree we should rework the object ownership model but would prefer to do that refactor separately.

Reply:
Fine by me 👍 .